File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n = matrix.length; //number of rows
        int m = matrix[0].length; // number of columns 
        int row =0; //Start at first row
        int col = m-1; // start at first column
        while(row<n && col >=0){
            if(matrix[row][col]==target){
                return true; //Found Target
            }else if(matrix[row][col]<target){
                row++; //Move Down
            }else{
                col--; //Move Left
            }
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    const n = matrix.length;
    const m = matrix[0].length;

    let row = 0;
    let col = m - 1;

    while (row < n && col >= 0) {
        const current = matrix[row][col];

        if (current === target) {
            return true;
        } else if (current < target) {
            row++;      // Move down
        } else {
            col--;      // Move left
        }
    }
    return false;
};